quickwit/quickwit-common/src/lib.rs

```diff
@@ -188,6 +188,32 @@ pub fn no_color() -> bool {
     matches!(env::var("NO_COLOR"), Ok(value) if !value.is_empty())
 }
 
+#[macro_export]
+macro_rules! assert_eventually {
+    ($cond:expr, $timeout:expr, $interval:expr) => {
+        let start = std::time::Instant::now();
+        loop {
+            if $cond {
+                break;
+            }
+            if start.elapsed() > $timeout {
+                panic!(
+                    "assertion failed: condition `{}` never became true within {} ms",
+                    stringify!($cond),
+                    $timeout.as_millis()
+                );
+            }
+            tokio::time::sleep($interval).await;
+        }
+    };
+    ($cond:expr, $timeout:expr) => {
+        assert_eventually!($cond, $timeout, std::time::Duration::from_millis(100));
```

Comment:
can we make the interval smaller? In the callsite the *timeout* was 1/4th of that already